smart_open/tests/test_s3.py

```diff
@@ -333,6 +333,24 @@ def test_nonexisting_bucket(self):
             with smart_open.s3.open('thisbucketdoesntexist', 'mykey', 'wb') as fout:
                 fout.write(expected)
 
+    def test_double_close(self):
+        s3 = boto3.resource('s3')
+        s3.create_bucket(Bucket='test_double_close')
+        text = u'там за туманами, вечными, пьяными'.encode('utf-8')
+        fout = smart_open.s3.open('test_double_close', 'key', 'wb')
+        fout.write(text)
+        fout.close()
+        fout.close()
+
+    def test_flush_close(self):
```

Comment:
what's an idea of this test?

Reply:
test_flush_close checks if we have a flush method.

test_double_close checks that calling close twice doesn't cause a crash.

Both of these tests were written as part of TDD. They capture the problematic behavior of smart open before the corresponding changes to s3 were made.